src/axom/inlet/SphinxDocWriter.hpp

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ *******************************************************************************
+ * \file SphinxDocWriter.hpp
+ *
+ * \brief This file contains the class definition of the SphinxDocWriter.
+ *******************************************************************************
+ */
+
+#ifndef INLET_SPHINXDOCWRITER_HPP
+#define INLET_SPHINXDOCWRITER_HPP
+
+#include <string>
+#include <vector>
+#include <fstream>
+
+#include "axom/sidre.hpp"
+#include "axom/inlet/DocWriter.hpp"
+
+namespace axom
+{
+namespace inlet
+{ 
+
+/*!
+ *******************************************************************************
+ * \class SphinxDocWriter
+ *
+ * \brief A DocWriter that is able write documentation in Sphinx RST format for 
+ * a given input deck.
+ *
+ * \see DocWriter
+ *******************************************************************************
+ */
+class SphinxDocWriter : public DocWriter {
+public: 
+  /*!
+  *******************************************************************************
+  * \brief A constructor for SphinxDocWriter.
+  * 
+  * \param [in] fileName The name of the file the documentation should be written to.
+  * 
+  * \param [in] sidreRootGroup The root of the sidre group that will be traversed
+  * to create documentation.
+  *
+  *******************************************************************************
+  */
+  SphinxDocWriter(const std::string& fileName, axom::sidre::Group* sidreRootGroup);
+
+  /*!
+  *******************************************************************************
+  * \brief Writes all documentation to a file.
+  * 
+  * This generates all RST-syntax documentation and writes it to the pre-specified
+  * file.
+  * 
+  * \param [in] sidreRootGroup The root of the sidre group that will be traversed
+  * to create documentation.
+  *
+  *******************************************************************************
+  */
+  void writeDocuments(axom::sidre::Group* sidreGroup);

```

Comment:
@aeshapar -- how is this `sidreGroup` parameter related to the `sidreRootGroup` class parameter?

Reply:
I realized there's actually no point in having that there since it would always be using the sidreRootGroup. Thanks for pointing it out!